### PR TITLE
adding sample_or_load function

### LINKED ISSUE #90: Add sample_or_load function
Add a function to load previous idata or sample the selected model.

### PR DESCRIPTION
**Pull request by CamSalda**
Adding sample_or_load function Closes #90

### CODE REVIEW

File: laos_gggi/sample.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import arviz as az
+import pymc as pm
+
+
+def sample_or_load(
+    fp: str,
+    *,
+    model: pm.Model = None,
+    resample: bool = False,
+    sample_kwargs: Optional[dict] = None,
```
Comment (jessegrabowski) [inline laos_gggi/sample.py:15]:
```suggestion
    sample_kwargs: dict | None = None,
```

File: laos_gggi/sample.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import os
+import pathlib
+import warnings
+from typing import Optional
```
Comment (jessegrabowski) [inline laos_gggi/sample.py:4]:
```suggestion

```

File: laos_gggi/sample.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        idata = az.from_netcdf(_fp)
+        idata.load()  # Force load to avoid mismatch if the memory is overwritten before idata is used
+    else:
+        if sample_kwargs is None:
+            sample_kwargs = {}
+        with model:
+            idata = pm.sample(**sample_kwargs)
+            idata = pm.sample_posterior_predictive(idata, extend_inferencedata=True)
+            idata = pm.compute_log_likelihood(idata, extend_inferencedata=True)
+            try:
```
Comment (jessegrabowski) [inline laos_gggi/sample.py:47]: I don't think we need this try/except. If it raises, you're just raising the error again anyway. Just let it fail loudly.

**Review by jessegrabowski — APPROVED** [review]
Small changes, pre-approving

File: laos_gggi/sample.py (hunk 3 above)
Comment (CamSalda) [reply laos_gggi/sample.py:46]: Sure.

**Pull request merged**